bundle/edu.gemini.ags/src/main/scala/edu/gemini/ags/impl/RadiusLimitCalc.scala

```diff
@@ -0,0 +1,86 @@
+    getAgsQueryRadiusLimits(guideProbe.getCorrectedPatrolField(ctx).asScalaOpt, ctx)
+
+  def getAgsQueryRadiusLimits(pf: Option[PatrolField], ctx: ObsContext): Option[RadiusConstraint] = {
+    // This gets a rectangle expressed in arcsec but in screen coordinates
+    // as if the base position were the upper left corner (0,0).
+    // x increases to the right, and y towards the bottom. :-(
+    // So it would need to be flipped around to work with intuitively,
+    // but we just need the distance from the base to the farthest corner
+    // of the probe limits.
+    val r2d = pf.map(_.outerLimitOffsetIntersection(ctx.getSciencePositions).getBounds2D) | new Rectangle2D.Double(0, 0, 0, 0)
+
+    // All the offset positions are far enough apart that the area of
+    // this rectangle is 0.  It's impossible to find any guide stars in
+    // range at all positions.  We'll just go ahead and do a search
+    // limited by the probe ranges as if there were no offset positions.
+    // It may find candidates, but will ultimately fail in the
+    // analysis.
+    (r2d.getWidth > 0 && r2d.getHeight > 0) option {
+      // We need the farthest corner, which will differ with port flips
+      // so just get the biggest absolute x and y.
+      val maxx = max(abs(r2d.getMinX), abs(r2d.getMaxX))
+      val maxy = max(abs(r2d.getMinY), abs(r2d.getMaxY))
+      val maxr = sqrt(maxx * maxx + maxy * maxy)
+
+      val maxAngle = Angle.fromArcsecs(maxr)
+      val minAngle = if (!r2d.contains(0, 0)) Angle.fromArcsecs(shortestDistance(r2d)) else Angle.zero
+
+      RadiusConstraint.between(maxAngle, minAngle)
+    }
+  }
+
+  private def shortestDistance(r: Rectangle2D): Double = {
+    val (minx, miny, maxx, maxy) = (r.getMinX, r.getMinY, r.getMaxX, r.getMaxY)
+
+    val dh = if (abs(miny) < abs(maxy)) shortestDistanceToHorizontalLine(new Line2D.Double(minx, miny, maxx, miny)) else shortestDistanceToHorizontalLine(new Line2D.Double(minx, maxy, maxx, maxy))
+    val dv= if (abs(minx) < abs(maxx)) shortestDistanceToVerticalLine(new Line2D.Double(minx, miny, minx, maxy)) else shortestDistanceToVerticalLine(new Line2D.Double(maxx, miny, maxx, maxy))
+    min(dh, dv)
+  }
+
+  private def shortestDistanceToHorizontalLine(l: Line2D): Double = {
+    val p1 = l.getP1
+    val p2 = l.getP2
+    if ((p1.getX <= 0) && (0 <= p2.getX)) {
+      // If the center falls between the horizontal extremes of the
+      // segment, it's the veritical distance to the line.
+      abs(p1.getY)
+    } else {
+      min(Zero.distance(p1), Zero.distance(p2))
+    }
+  }
+
+  private def shortestDistanceToVerticalLine(l: Line2D): Double = {
+    val p1 = l.getP1
+    val p2 = l.getP2
+    if ((p1.getY <= 0) && (0 <= p2.getY)) {
+      // If the center falls between the vertical extremes of the
+      // segment, it's the horizontal distance to the line.
+      abs(p1.getX)
+    } else {
+      min(Zero.distance(p1), Zero.distance(p2))
+    }
+  }
```

Comment:
Seems like we might as well collapse these two `shortestDistance` into one passing in the `Point2D => Double` function to use?